src/Symfony/Component/Translation/MessageCatalogue.php

```diff
@@ -272,4 +292,16 @@ private function addMetadata(array $values)
             }
         }
     }
+
+    /**
+     * Get default domain.
+     *
+     * @param string $domain
+     *
+     * @return string
+     */
+    private function getDefaultDomain($domain)
+    {
+        return $domain ?? TranslatorInterface::DEFAULT_DOMAIN;
```

Comment:
I'd still prefer `public function __construct($locale, array $messages = array(), $defaultDomain = 'messages')` without the constant.

Reply:
that would mean the default domain can be different per instance and translator and catalog etc which makes it impossible to rely on the default domain as you don't know what will be used.